packages/dd-trace/test/appsec/sdk/set_user.spec.js

```diff
@@ -0,0 +1,156 @@
+'use strict'
+
+const proxyquire = require('proxyquire')
+const agent = require('../../plugins/agent')
+const tracer = require('../../../../../index')
+const appsec = require('../../../src/appsec')
+const Config = require('../../../src/config')
+const getPort = require('get-port')
+const axios = require('axios')
+
+describe('setUser', () => {
+  describe('Check internal callings', () => {
+    const tracer = {}
+    let sdk, mockSetTag, mockRootSpan, getRootSpan
+
+    beforeEach(() => {
+      mockSetTag = sinon.stub()
+      mockRootSpan = {
+        context: () => {
+          return { _tags: { 'usr.id': 'mockUser' } }
+        },
+        setTag: mockSetTag }
+      getRootSpan = sinon.stub().returns(mockRootSpan)
+
+      const { setUser } = proxyquire('../../../src/appsec/sdk/set_user', {
+        './utils': { getRootSpan }
+      })
+
+      const AppsecSdk = proxyquire('../../../src/appsec/sdk', {
+        './set_user': { setUser }
+      })
+
+      sdk = new AppsecSdk(tracer)
+    })
+
+    it('setUser should call setTag with proper values', () => {
+      const user = { id: 'user' }
+      sdk.setUser(user)
+      expect(mockSetTag).to.be.calledOnceWithExactly('usr.id', 'user')
+    })
+
+    it('setUser should not call setTag when no user is passed', () => {
+      sdk.setUser()
+      expect(mockSetTag).not.to.have.been.called
+    })
+
+    it('setUser should not call setTag when user is empty', () => {
+      const user = {}
+      sdk.setUser(user)
+      expect(mockSetTag).not.to.have.been.called
+    })
+
+    it('setUser should call setTag with every attribute', () => {
+      const user = {
+        id: '123',
+        email: 'a@b.c',
+        custom: 'hello'
+      }
+
+      sdk.setUser(user)
+      expect(mockSetTag).to.have.been.calledThrice
+      expect(mockSetTag.firstCall).to.have.been.calledWithExactly('usr.id', '123')
+      expect(mockSetTag.secondCall).to.have.been.calledWithExactly('usr.email', 'a@b.c')
+      expect(mockSetTag.thirdCall).to.have.been.calledWithExactly('usr.custom', 'hello')
+    })
+  })
+
+  describe('Check internal callings, no rootSpan', () => {
+    const tracer = {}
+    const getRootSpan = sinon.stub().returns(undefined)
+    const setUserTagsStub = sinon.stub()
+
+    const { setUser } = proxyquire('../../../src/appsec/sdk/set_user', {
+      './utils': { getRootSpan },
+      setUserTags: setUserTagsStub
+    })
+
+    const AppsecSdk = proxyquire('../../../src/appsec/sdk', {
+      './set_user': { setUser }
+    })
+
+    const sdk = new AppsecSdk(tracer)
```

Comment:
Usually, this type of initializations are done in `beforeEach` blocks
```
let tracer
beforeEach(() => {
  tracer = ...
}
```

Reply:
this whole describe() could be just moved in the describe above, and this specific test would change the rootSpan stub to return undefined, instead of duplicating all that boilerplate